python/lsst/ip/isr/linearize.py

```diff
@@ -127,32 +139,280 @@ def fromYaml(self, yamlObject):
+            outDict['tableData'] = self.tableData.tolist()
+
+        return outDict
+
+    @classmethod
+    def readText(cls, filename):
+        """Read linearity from text file.
+
+        Parameters
+        ----------
+        filename : `str`
+            Name of the file containing the linearity definition.
+        Returns
+        -------
+        linearity : `~lsst.ip.isr.linearize.Linearizer``
+            Linearity parameters.
+        """
+        data = ''
+        try:
+            with open(filename, 'r') as f:
+                data = yaml.load(f, Loader=yaml.FullLoader)
+        except Exception:
+            def yamlLoad(x):
+                return yaml.load(x, Loader=yaml.Loader)
+            with open(filename, 'r') as f:
+                data = yamlLoad(f)
+        return cls().fromYaml(data)
+
+    def writeText(self, filename):
+        """Write the linearity model to a text file.
+
+        Parameters
+        ----------
+        filename : `str`
+            Name of the file to write.
+
+        Returns
+        -------
+        used : `str`
+            The name of the file used to write the data.
+
+        Notes
+        -----
+        The file is written to YAML format and will include any metadata
+        associated with the `Linearity`.
+        """
+        outDict = self.toDict()
+
+        with open(filename, 'w') as f:
+            yaml.dump(outDict, f)
+
+        return filename
+
+    @classmethod
+    def readFits(cls, filename):
+        """Read linearity from a FITS file.
+
+        Parameters
+        ----------
+        filename : `str`
+            Name of the file containing the linearity definition.
+        Returns
+        -------
+        linearity : `~lsst.ip.isr.linearize.Linearizer``
+            Linearity parameters.
+        """
+        table = afwTable.BaseCatalog.readFits(filename)
+        metadata = table.getMetadata()
+
+        linDict = dict()
+        linDict['metadata'] = metadata
+        linDict['detectorId'] = metadata['DETECTOR_ID']
+        linDict['detectorName'] = metadata['DETECTOR_NAME']
+        try:
+            linDict['detectorSerial'] = metadata['DETECTOR_SERIAL']
+        except Exception:
+            linDict['detectorSerial'] = 'NOT SET'
+        linDict['amplifiers'] = dict()
+
+        for row in table:
+            record = row.extract("*")
+            ampName = record['AMPLIFIER_NAME']
+            ampDict = dict()
+            ampDict['linearityType'] = record['TYPE']
+            ampDict['linearityCoeffs'] = record['COEFFS']
+            ampDict['linearityBBox'] = Box2I(Point2I(record['BBOX_X0'], record['BBOX_Y0']),
+                                             Extent2I(record['BBOX_DX'], record['BBOX_DY']))
+
+            linDict['amplifiers'][ampName] = ampDict
+
+        try:
+            table = afwTable.BaseCatalog.readFits(filename, 2)
+            tableData = []
+            for row in table:
+                record = row.extract('*')
```

Comment:
Again. My fault but this has to be changed for performance. It could now probably look like:
```python
linDict["tableData"] = [record[lookupValuesKey] for record in table]
```